sklearn/manifold/locally_linear.py

```diff
@@ -289,7 +289,14 @@ def locally_linear_embedding(
         raise ValueError("unrecognized method '%s'" % method)
 
     nbrs = NearestNeighbors(n_neighbors=n_neighbors + 1, n_jobs=n_jobs)
+
     nbrs.fit(X)
+
+    if isinstance(X, (BallTree, KDTree, NearestNeighbors)) and \
+       not (X.metric == 'euclidean'
+            or (X.metric == 'minkowski' and X.p == 2)):
+        raise ValueError("metric must be euclidean or equivalent")
```

Comment:
Best practice: put the offending value in the exception. I tend to like error messages like this.

```py
"'parameter' should be have this value for this and this reason. Got 'parameter={}' instead.".format(value)
```

Reply:
Fixed